preview/FS-1068-open-type-declaration.md

```diff
@@ -0,0 +1,273 @@
+
+```fsharp
+open type byref<int> // Compile error
+```
+
+### Tooling
+
+Updates to FCS are made to understand `open type` to display a list of types that can be opened.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+* This introduces another avenue to encounter issues when resolving overloads
+* This introduces more avenues to mix overloaded members with type inference, which can lead to source breaking changes if APIs defining those members add overloads over time
+* Code using this feature could be harder to understand without editor tooling
+
+## Alternatives
+[alternatives]: #alternatives
+
+###  `open C`, `open type C` or `open static C`
+
+We have resolved to use `open type C`.
+
+Issues to do with potential breaking changes do to changes in shadowing led to a change of heart, see https://github.com/fsharp/fslang-design/issues/352#issuecomment-522533863
+
+See original discussion here: https://github.com/fsharp/fslang-design/issues/352#issuecomment-499146012.
+
+### Combining overloaded methods from different types
+
+We will **not** allow combination of method overloads from different types according to C# rules. Scenarios such as FSI or Jupyter Notebooks: we need to be able to shadow method groups in order to not have old method overloads from previous types in scope.
+
+When multiple methods of the same name are in scope, they can be overloaded provided that their signatures are unique. If we did, this is an example:
+
+```fsharp
+type A() =
+    static member M(x: int) = x * 2
+
+type B() =
+    static member M(x: float) = x * 2.0
+
+open type A
+open type B
+
+// Both methods are resolved
+M(1)
+M(1.0)
+```
+
+If the signatures for `M` were not unique, then it would not be possible to call `M`. Full qualification is required, or if the source is available, a change to one of the methods is required:
+
+```fsharp
+type A() =
+    static member M(x: int) = x * 2
+
+type B() =
+    static member M(x: int) = x * 2
+
+open type A
+open type B
+
+// Error: ambiguous call
+M(1)
+```
+
+Because overloads coming from multiple opened types are not resolved in the context of a type, we would not require compatibility with the existing rules for resolution.
+
+Because APIs utilizing static members generally use methods instead of properties, we prefer methods over properties and adjust the previous ordered-list as such:
+
+Try to resolve `member-ident` to one of the following, in order:
+
+1. A union case.
+2. **A method group.**
+3. A property group.
+4. A field.
+5. An event.
+6. **A method group of extension members, by consulting the `ExtensionsInScope` table.**
+7. A property group of extension members , by consulting the `ExtensionsInScope` table.
+8. A nested type `type-nested`. Recursively resolve .rest if it is present, otherwise return `type-nested`
+
+That is to say, we will resolve methods over properties in the context of opening static classes or static members extending a static class.
```

Comment:
This section reads as if we're going to do this. Should either remove it or reword it to say that it's the avenue we _would_ take.